src/wallet/wallet.cpp

```diff
@@ -2529,6 +2537,7 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                 BOOST_FOREACH (const CRecipient& recipient, vecSend)
                 {
                     CTxOut txout(recipient.asset, recipient.nAmount, recipient.scriptPubKey);
+                    txout.nNonce.vchCommitment = std::vector<unsigned char>(recipient.confidentiality_key.begin(), recipient.confidentiality_key.end());
```

Comment:
I think I have to do this for change outputs as well, otherwise we'll not actually blind the change output in a follow-up `blindtransaction` call.